test/libsolidity/smtCheckerTests/bmc_coverage/branches_with_return/branches_in_modifiers.sol

```diff
@@ -0,0 +1,27 @@
+pragma experimental SMTChecker;
+
+contract C {
+
+	uint x;
+
+	modifier check() {
+		require(x == 0);
+		_;
+		assert(x == 1); // should fail;
+		assert(x == 0); // should hold;
+	}
+
+	modifier inc() {
+		if (x == 0) {
+			return;
+		}
+		x = x + 1;
+		_;
+	}
+
+	function test() check inc public {
```

Comment:
Can you also add  some return inside the function to make it more complicated

Reply:
Added more complicated example `branches_in_modifiers.sol` There was actually a bug detected already with this test. There was an assertion failure that was not detected before. Now it should be.